enhancements/installer/azure-support-known-cloud-environments.md

```diff
@@ -0,0 +1,231 @@
+
+### Goals
+
+1. Install cluster to known Azure clouds.
+
+### Non-Goals
+
+1. Supporting custom Azure clouds deployments that are not defined in Azure SDKs, i.e. custom BaseURI for endpoints will not suported.
+2. Supporting self-signed server certificates for endpoints specifically.
+
+## Proposal
+
+The installer will allow the users to specify the name of the Azure cloud using the install-config.yaml. The value will be restricted to one of the known [names](https://github.com/Azure/go-autorest/blob/9132adfc9db3007653dccb091b6a6c8e09b74ef5/autorest/azure/environments.go#L34-L39). When no cloud is specified the installer will default to `AzurePublicCloud`. The installer will use the endpoints pre-defined in the Azure SDK for the specified clouds to communicate with Azure APIs to perform validations, create machineset objects, and configuring the terraform `azurerm` provider.
+
+When using different Azure cloud, the cluster operators also need the discover the information, therefore, the installer will make these available using the config.openshift.io/v1 Infrastructure object. The cluster operators can then use Azure cloud name from the object to configure the Azure SDK to use the corresponding endpoints for communications.
+
+### User Stories [optional]
+
+Detail the things that people will be able to do if this is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+
+#### US1: Installer cluster in MAG
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Infrastructure global configuration
+
+Since almost of the cluster operators that communicate to the the Azure API need the use the Azure cloud provided by the user, the Infrastructure global configuration seems like the best place to store and discover this information.
+
+The Infratructure object contains user-editable platform configurable in `spec`, and discovery information that is not user-editable in `status`. The Azure cloud is a one-time setting that the user shouldn't be allowed to change and therefore, this information will only be allowed in the `status` section.
+
+```go
+// AzurePlatformStatus holds the current status of the Azure infrastructure provider.
+type AzurePlatformStatus struct {
+	// resourceGroupName is the Resource Group for new Azure resources created for the cluster.
+	ResourceGroupName string `json:"resourceGroupName"`
+
+
+	// networkResourceGroupName is the Resource Group for network resources like the Virtual Network and Subnets used by the cluster.
+	// If empty, the value is same as ResourceGroupName.
+	// +optional
+	NetworkResourceGroupName string `json:"networkResourceGroupName,omitempty"`
+    
+	// cloudName is the name of the Azure cloud environment which can be used to configure the Azure SDK
+	// with the appropiate Azure API endpoints.
+	// If empty, the value is equal to `AzurePublicCloud`.
```

Comment:
nit: `If empty, consumers should default to AzurePublicCloud.` or some such?  If it's empty, the value here will be empty, right?

Reply:
done